lnwire/dyn_ack.go

```diff
@@ -62,31 +60,27 @@ func (da *DynAck) Encode(w *bytes.Buffer, _ uint32) error {
 		return err
 	}
 
-	var tlvRecords []tlv.Record
-	da.LocalNonce.WhenSome(func(nonce Musig2Nonce) {
-		tlvRecords = append(
-			tlvRecords, tlv.MakeStaticRecord(
-				DALocalMusig2Pubnonce, &nonce,
-				musig2.PubNonceSize, nonceTypeEncoder,
-				nonceTypeDecoder,
-			),
-		)
-	})
-	tlv.SortRecords(tlvRecords)
-
-	tlvStream, err := tlv.NewStream(tlvRecords...)
+	// Create extra data records.
+	producers, err := da.ExtraData.RecordProducers()
 	if err != nil {
 		return err
 	}
 
-	var extraBytesWriter bytes.Buffer
-	if err := tlvStream.Encode(&extraBytesWriter); err != nil {
+	// Append the known records.
+	da.LocalNonce.WhenSome(
+		func(rec tlv.RecordT[tlv.TlvType14, Musig2Nonce]) {
+			producers = append(producers, &rec)
+		})
```

Comment:
Style nit: can add a comma here for the line folding.

Reply:
Updated